R/configuration-plan.R

```diff
@@ -0,0 +1,117 @@
+#' @title ConfigurationPlan
+#' @docType class
+#' @description  Abstract class for ConfigurationPlan
+#' @export
+ConfigurationPlan <- R6::R6Class(
+  "ConfigurationPlan",
+  public = list(
+    reInputPath = NULL,
+    reOutputPath = NULL,
+    simulationMapping = NULL,
+    observedDataSets = NULL,
+    plots = NULL,
+    inputs = NULL,
+    sections = NULL,
+    intro = NULL,
+
+    initialize = function(jsonFile,
+                              reInputPath,
+                              reOutputPath,
+                              overwriteREOutputPath = TRUE) {
+      print(paste(
+        "Reporting Engine: Initilization of Configuration Plan from",
+        jsonFile
+      ))
+      configurationPlan <- jsonlite::fromJSON(jsonFile)
+      self$reInputPath <- reInputPath
+      self$reOutputPath <- reOutputPath
+
+      if (dir.exists(reOutputPath)) {
+        warning(paste("Reporting Engine output path:", reOutputPath, "already exists"),

```

Comment:
@pchelle CAn we create a file like in TLF with all the constant? (messages.R or whatever we called it)? 
That way, if we need to correct a typo, we know where to look